### PR TITLE
add argument

### PR DESCRIPTION
**Pull request by cxxxr**

### CODE REVIEW

File: Sources/Pokepay/Pokepay.swift [hunk 1]
```diff
@@ -301,8 +301,12 @@ public struct Pokepay {
             }
         }
 
-        public func getAuthorizationUrl() -> String {
-            return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+        public func getAuthorizationUrl(contact: String? = nil) -> String {
+            if contact != nil {
+                return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code&contact=\(contact)"
```
Comment (fukamachi) [inline Sources/Pokepay/Pokepay.swift]: [must] `contact` should be URL-escaped.

**Comment by smihica** [comment]
Please add test.

File: PokepayTests/PokepayTests.swift [hunk 2]
```diff
@@ -269,11 +269,9 @@ AQIDAQAB
     }
 
     func testAuthorizationUrl() {
-        let expect = expectation(description: "OAuthClient.getAuthorizationUrl")
         let oauth = Pokepay.OAuthClient(clientId: "3qyJZlDnJbGK5roa-5XLkw", clientSecret: "kz96I8SDmjl2x77aAI9iNvAjc0cneL3UoK6zKLjmdXwmoghC7FSRrqKr")
-        print(oauth.getAuthorizationUrl())
-        expect.fulfill()
-        waitForExpectations(timeout: 5.0, handler: nil)
+        XCTAssertEqual(oauth.getAuthorizationUrl(), "https://www-dev.pokepay.jp/oauth/authorize?client_id=3qyJZlDnJbGK5roa-5XLkw&response_type=code")
+        XCTAssertEqual(oauth.getAuthorizationUrl(contact: "09012345678"), "https://www-dev.pokepay.jp/oauth/authorize?client_id=3qyJZlDnJbGK5roa-5XLkw&response_type=code&contact=09012345678")
```
Comment (fukamachi) [inline PokepayTests/PokepayTests.swift]: Can you add an E-mail case to make sure if its URL-encoded?

File: Sources/Pokepay/Pokepay.swift [hunk 3]
```diff
@@ -301,8 +301,12 @@ public struct Pokepay {
...
         }
 
-        public func getAuthorizationUrl() -> String {
-            return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+        public func getAuthorizationUrl(contact: String? = nil) -> String {
+            if contact != nil {
+                return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code&contact=\(contact!)"
+            } else {
+                return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+            }
```
Comment (nicchi--1985) [inline Sources/Pokepay/Pokepay.swift]: comment test

File: Sources/Pokepay/Pokepay.swift [hunk 4]
```diff
@@ -301,8 +301,15 @@ public struct Pokepay {
...
-            return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+        public func getAuthorizationUrl(contact: String? = nil) -> String {
+            let url = "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+            if contact != nil {
+                let encodedContact = contact!.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed)
+                precondition(encodedContact != nil)
+                return "\(url)&contact=\(encodedContact!)"
+            } else {
+                return url
+            }
```
Comment (fukamachi) [inline Sources/Pokepay/Pokepay.swift]: [should] Checking `Optional` value with `if` can be replaced with `guard ... else` which is thought to be better style in some documents for readability.

Like,

```swift
Using a guard statement for requirements improves the readability of your code, compared to doing the same check with an if statement. It lets you write the code that’s typically executed without wrapping it in an else block, and it lets you keep the code that handles a violated requirement next to the requirement.
```

at [Early Exits of the official Swift language guide](https://docs.swift.org/swift-book/LanguageGuide/ControlFlow.html#ID525),  or,

```swift
guards for Early Exits

A guard statement, compared to an if statement with an inverted condition, provides visual emphasis that the condition being tested is a special case that causes early exit from the enclosing scope.
```

in [Swift Style Guide](https://google.github.io/swift/#guards-for-early-exits) by Google.


```suggestion
            guard let contact = contact else {
                return url
            }

            let encodedContact = contact.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed)
            precondition(encodedContact != nil)
            return "\(url)&contact=\(encodedContact!)"
```

File: Sources/Pokepay/Pokepay.swift [hunk 5]
```diff
@@ -301,8 +301,14 @@ public struct Pokepay {
             }
         }
 
-        public func getAuthorizationUrl() -> String {
-            return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+        public func getAuthorizationUrl(contact: String? = nil) -> String {
+            let url = "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+            guard let contact = contact else {
+                return url;
```
Comment (fukamachi) [inline Sources/Pokepay/Pokepay.swift]: [must] Unnecessary semicolon.

**Comment by smihica** [comment]
Please update version after merging @cxxxr

**Pull request merged**